### PR TITLE
Doc/library/os.rst: `os.waitid` absent on MacOS

### PR DESCRIPTION
**Pull request by johnhawkinson**
`os.waitid()` is disabled on MacOS, and that's not documented.
It's not precisely clear why its disabled, and it was discussed in #55021 [(deeplink)](https://github.com/python/cpython/issues/55021#issuecomment-1093530180) which was closed without resolution.

But regardless of why, the documentation needs to clearly reflect that the function is not available.

I've followed the model of `os.fdatasync()` and noted the lack of availability in the `Note` section, rather than the `Availability` section. Absent that precedent, I would have instead changed `Availability` to ` Unix, not Emscripten, not Macos, not WASI`, perhaps somewhat consistent with `CLOCK_UPTIME_RAW` in [time.rst L866](https://github.com/python/cpython/blob/5e9f471e7db30893fb3f42681f17fdcdb70069ee/Doc/library/time.rst?plain=1#L866-L866).

I hope this is appropriate for a PR without an Issue, but if I'm wrong, I apologize and am happy to file an Issue.

----
:books: Documentation preview :books:: https://cpython-previews--104558.org.readthedocs.build/

### CODE REVIEW

**Comment by johnhawkinson** [comment]
> `os.waitid()` is disabled on MacOS, and that's not documented.

I probably should have supported the statement that it's disabled, sorry.
As discussed in the linked issue, the disabling is from the `&& !defined(__APPLE__)` here:

https://github.com/python/cpython/blob/c649df63e0d052044a4660101d5769ff46ae9234/Modules/posixmodule.c#L9191-L9213

File: Doc/library/os.rst
Comment (aisk) [inline Doc/library/os.rst]:
```suggestion
      This function is not available on macOS.
```

IMO, it spelles as `macOS` in the rest of this document.

Comment (aisk) [reply Doc/library/os.rst]: Can we just change the exist `availability` section up here?

Comment (johnhawkinson) [reply Doc/library/os.rst]: @aisk: On spelling, the document is not consistent and it appears both ways. As I noted in the description above:
> I've followed the model of os.fdatasync()

and that is both as to the spelling/casing ("MacOS" vs. "macOS") as well as the Note section rather than the Availability section. It is not how I would have written it without that precedent.

That said, I'm happy to conform to whatever standard, but the document style is inconsistent, and I'd really just like the semantics fixed before the next release, I don't care about the style. I'd tend to think the style correction should be a different PR that works harder to make everything consistent, but I'm not involved enough in Python development to know the proper cultural processes.

**Comment by ronaldoussoren** [comment]
The more interesting question here is why `os.waitid` is not exposed on macOS.

There's some discussion in https://github.com/python/cpython/issues/55021, `waitid` was not exposed for unclear reasons. IMHO we should expose it for macOS and if needed document limitations with the API (it was disabled for unspecified problems in versions of the OS that we no longer supported in our installers).

**Comment by johnhawkinson** [comment]
> The more interesting question here is why `os.waitid` is not exposed on macOS.

You know…no it's not!

If it takes 6 months to get a doc fix in and 12 months to have a policy debate and maybe change the policy and get a code fix in, I am entirely uninterested in postponing the former in favor of the latter. Open a new PR or Issue to expose `waitid`  but don't hold up fixing the broken documentation now.

Thank you. I am not as bitter as I sound.

**Comment by ronaldoussoren** [comment]
> 
> Thank you. I am not as bitter as I sound.

I don't know about other core devs, but personally I only regularly check GitHub issues for things to work on. That's mostly out of habit, but we do have a longstanding culture of having issues for almost all changes.  I'm not going to argue that's a good thing, but it does explain why PRs without an issue can get lost and not get core dev attention.

Anyway, I'll file an issue about exposing `os.waitid` in 3.13 and will return to this PR soon with the intention to merge it.

File: Doc/library/os.rst
Comment (ronaldoussoren) [reply Doc/library/os.rst]: The correct spelling is "macOS" according to Apple. I'll fix the spelling in the `fdatasync` documentation in a follow-up PR.

Documenting this in a note is fine, as you note we do this in more places.

**Review by ronaldoussoren — APPROVED** [review]
LGTM

Thanks for the PR!

**Pull request merged**
